barretenberg/cpp/src/barretenberg/ecc/groups/element_impl.hpp

```diff
@@ -901,6 +901,28 @@ std::vector<affine_element<Fq, Fr, T>> element<Fq, Fr, T>::batch_mul_with_endomo
             /*finite_field_additions_per_iteration=*/7,
             /*finite_field_multiplications_per_iteration=*/6);
     };
+
+    // Since the core algorithm has no edgecase detection, it can't handle -1. (Because constructing -1 is r + -1 as
```

Comment:
Can you make this comment more explicit please?

Reply:
Done